packages/mui-joy/src/AutocompleteOption/AutocompleteOption.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import { OverridableComponent } from '@mui/types';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import composeClasses from '@mui/base/composeClasses';
+import { ListItemButtonRoot } from '../ListItemButton/ListItemButton';
+import { styled, useThemeProps } from '../styles';
+import { getAutocompleteOptionUtilityClass } from './autocompleteOptionClasses';
+import { AutocompleteOptionOwnerState, AutocompleteOptionTypeMap } from './AutocompleteOptionProps';
+
+const useUtilityClasses = (ownerState: AutocompleteOptionOwnerState) => {
+  const { color, variant } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      color && `color${capitalize(color)}`,
+      variant && `variant${capitalize(variant)}`,
+    ],
+  };
+
+  return composeClasses(slots, getAutocompleteOptionUtilityClass, {});
+};
+
+export const AutocompleteOptionRoot = styled(ListItemButtonRoot as unknown as 'li', {
```

Comment:
Instead of re-using components, should we maybe re-use hooks? This will create a cascading updates (when someone customizes the `ListItemButton` it will reflect on the `Autocomplete` component. I am not saying this is a bad things, but maybe it's something worth discussing first, before implementing it everywhere.

Reply:
More over, now we will have the class names for both components here.

Reply:
> when someone customizes the ListItemButton it will reflect on the Autocomplete component

Yes, that's the intention. This is similar to Material UI's MenuItem which reuses the ListItemButton.

> More over, now we will have the class names for both components here.

We won't because we are reusing the styled-component, not the whole `ListItemButton`. This is a better abstraction because the `AutocompleteOption` does not need any logic inside the `ListItemButton`, e.g. `useButton` hook.

Reply:
Thanks for the explanation.

Reply:
I pushed an improvement to this from your point "This will create a cascading updates (when someone customizes the ListItemButton it will reflect on the Autocomplete component"

I think it makes sense. The more inheritance, the more complex for theming. My updates let components reuse styles from other components without inheriting theming capability.